src/pyedb/configuration/cfg_components.py

```diff
@@ -51,12 +51,13 @@ def __init__(self, **kwargs):
 
 
 class CfgComponent(CfgBase):
-    protected_attributes = ["reference_designator"]
+    protected_attributes = ["reference_designator", "definition", "location", "angle", "placement_layer"]
```

Comment:
After some though I'm wondering if we shouldn't drop the `protected_attributes` approach.
It's a bit confusing to ask for attributes and not retrieving the whole attributes but a subset of them without knowing that there is such filter (from the method name).

Since it's mainly associated to the current `get_attributes` and `set_attributes`, why not reworking this to use:
- `to_dict` (equivalent to `get_attributes`) but without exclude.
- `update` (equivalent to `set_attributes`) where we would have a call to `to_dict` with inner filtering as we don't want to update certain attributes.

That way the `protected_attributes` won't be a class attribute that is accessible by the user. The attributes we don't want them to update would be set in stone inside `update` and we could trigger warnings when one wants to do it.
What do you think ? If you agree, I can help with that.

Reply:
@SMoraisAnsys  Great idea. I will change according to your suggestion in a separate PR.

Reply:
> @SMoraisAnsys Great idea. I will change according to your suggestion in a separate PR.

I was supposed to work on that at some point (if I'm not wrong) while leveraing `pydantic`.
Mind if I do it as planned ? Sorry for the delay in handling that though !